src/core/dense_set.h

```diff
@@ -187,6 +187,14 @@ class DenseSet {
       return curr_entry_->HasTtl() ? owner_->ObjExpireTime(curr_entry_->GetObject()) : UINT32_MAX;
     }
 
+    void SetExpiryTime(uint32_t ttl_sec) {
```

Comment:
And this can be moved directly to `StringMap`. If you look on `hset_family` we use `StringMap *`, so we don't really need to implement this in `DenseSet` class.

Reply:
Why wouldn't we/you want to have `SetExpiryTime` in `DenseSet` where that's where everything else related to TTL is handled?

`StringSet` also relies on access to `HasExpiry()` here, because the ttl bits get stripped when calling `GetObject()`. Maybe there's a way to avoid that, but it seems like that would have a cascading effect.

Reply:
+1 to @NegatioN . We use expiry for both string_set and string_map so it's better to keep it as generic as possible